### ISSUE TITLE
xAxes ticks not updated after new data

### ISSUE THREAD

**Issue by NicoP-S** [victorgarciaesgi/vue-chart-3#80]
## Describe the bug

Start: two day range hourly = 48 data elements & labels  -> everything OK
![image](https://user-images.githubusercontent.com/8988747/149532434-0740e3a9-4b55-425d-a852-f783664438b3.png)

Switching to day basis = two data elements and two labels.
The data is updated but the chart xAxes does not. I tried `this.$refs.chart.chartInstance.update()` but nothing changed
I thought the data i put into the chart is not reactive. So i added an output of the chartData in the template and the data is updating.

![image](https://user-images.githubusercontent.com/8988747/149532694-a3f8a9c9-1645-42e0-bd4a-5fd4e3ce116b.png)

I checked the ticks with the following
``` 
console.log("Count Labels",this.chartData.labels.length);
console.log("Count Data",this.chartData.datasets[0].data.length);
console.log("Count cAxis ticks",this.$refs.chart.chartInstance.scales.xAxes.ticks.length)

------ FIRST CHART
Count Labels 48
Count Data 48
Count cAxis ticks 24

----- SECOND CHART
Count Labels 2
Count Data 2
Count cAxis ticks 24
```

Is there a way to force update of the xAxes ticks?

## Version of `vue-chart-3`
v2.0.5*

## Version of Vue

- [ ] Vue 3
- [X] Vue 2
- [ ] Nuxt 2
- [ ] Nuxt 3

**Comment by mgregory73** [comment]
I am experiencing the same issue with the labels not reflecting the data after a change. Seems if you interact with the chart (i.e., selecting a bar) it  corrects, so maybe it's a nextTick thing. Anyhow, reverting to 0.5.11 fixes it for now. That said, this project is great and a real life saver. Looking forward to a fix. Thanks so much and keep up the great work.

**Comment by victorgarciaesgi** [comment]
Hi! Sorry for the late reply I don't have much time to look into this. If I find a little spot i swear to look at it! :D If you want to make it easier can you make a codesanbox with your bug? :D

**Comment by mgregory73** [comment]
Hi Victor,

Again many thanks for the work you've done here. 

I don't know if this helps you but the change that causes the labels not to update happens in fix #33,   
https://github.com/victorgarciaesgi/vue-chart-3/commit/b3b6769c7027493bd0fb7e4b10cee5c0f56888a2#

Specifically, replacing `renderChart()` at line 151/142 of components.ts with  `chart.data.datasets = newData.datasets` seems to be related. Reverting it locally fixes the issue for me. There are a lot of other refactors in there so I'm not sure that is the simple solution, but it might point you in the right direction. 

Cheers

**Comment by NicoP-S** [comment]
Hi,
i tried to create a repo which shows the error but did't get it replicated.
I checked the changes of commit https://github.com/victorgarciaesgi/vue-chart-3/commit/b3b6769c7027493bd0fb7e4b10cee5c0f56888a2 and tried to undo them in my code. 

After replacing https://github.com/victorgarciaesgi/vue-chart-3/blob/af4d6b31434cd80f0410aa1f139e34e5b0c9b1aa/src/components.ts#L133  with the following code it works...

node_modules/vue-chart-3/dist/components.js
```
if (chart) {
     handleChartDestroy();
}
renderChart();
```

In my app I use the options api. 
Instead of `chart.data.datasets = newData.datasets;` 
i would do something like `this.$set(chart.data,'datasets',newData.datasets)`
Could this be the problem?

**Comment by NicoP-S** [comment]
@victorgarciaesgi  i got an running codesandbox now
https://codesandbox.io/s/vue-chart-3-vue-2-composition-api-forked-4mf88?file=/src/App.vue

**Comment by victorgarciaesgi** [comment]
Sorry for responding just now. I will fix it!

**Comment by NicoP-S** [comment]
i updated to latest @vue/composition-api 1.4.6... but the error is still there

**Comment by victorgarciaesgi** [comment]
Hmm try cleaning node_modules and cache, it's not the first down this error emerges for me and others, and the fix was always to update the `@vue/composition-api` version or remove/reinstall :/

**Comment by NicoP-S** [comment]
Ok after delete package-lock.json it is running again. But the graph is still not correct.

I updated my codesandbox too: 
https://codesandbox.io/s/vue-chart-3-vue-2-composition-api-forked-4mf88?file=/src/main.js

**Comment by victorgarciaesgi** [comment]
Thanks for the sandbox! I will look at it :)

**Comment by victorgarciaesgi** [comment]
It should be the same issue as #108 in Vue 3. I will look at it

**Comment by NicoP-S** [comment]
@victorgarciaesgi is the fix from issue 108 also working for vue2?

**Comment by victorgarciaesgi** [comment]
Hum no sorry i overlooked, #108 was a Vue 3 problem, but it should be related, the fix should be the same

**Comment by victorgarciaesgi** [comment]
Should be fixed in `2.1.3`!

**Comment by NicoP-S** [comment]
@victorgarciaesgi i just tested the codepen and my own codebase and it's working now :)
Thank you!
